app/assets/javascripts/bastion/components/notification.service.js

```diff
@@ -16,8 +16,11 @@ angular.module('Bastion.components').service("Notification", ['$interpolate', 'f
         return result;
     }
 
-    this.setSuccessMessage = function (message, context) {
-        foreman.toastNotifications.notify({message: interpolateIfNeeded(message, context), type: 'success'});
+    this.setSuccessMessage = function (message, options) {
+        if (options === angular.isUndefined) {
+            options = {link: ""};
+        }
+        foreman.toastNotifications.notify({message: interpolateIfNeeded(message, options.context), type: 'success', link: options.link});
```

Comment:
@johnpmitsch I'm getting the following error for it:
```
TypeError: Cannot read property 'context' of undefined
    at Object.setSuccessMessage (bastion-46c70c6bc8dbfb49bd30f57d8ec38039cf096443284e5bd4a0dac8a55ac8737d.js:87515)
    at success (bastion_katello-638b1c206284a094f9567ef353eed3374f77bbf130771fc42a829119c85fb8a5.js:14516)
    at bastion-46c70c6bc8dbfb49bd30f57d8ec38039cf096443284e5bd4a0dac8a55ac8737d.js:48330
    at processQueue (bastion-46c70c6bc8dbfb49bd30f57d8ec38039cf096443284e5bd4a0dac8a55ac8737d.js:32696)
    at bastion-46c70c6bc8dbfb49bd30f57d8ec38039cf096443284e5bd4a0dac8a55ac8737d.js:32712
    at Scope.$eval (bastion-46c70c6bc8dbfb49bd30f57d8ec38039cf096443284e5bd4a0dac8a55ac8737d.js:33964)
    at Scope.$digest (bastion-46c70c6bc8dbfb49bd30f57d8ec38039cf096443284e5bd4a0dac8a55ac8737d.js:33780)
    at Scope.$apply (bastion-46c70c6bc8dbfb49bd30f57d8ec38039cf096443284e5bd4a0dac8a55ac8737d.js:34072)
    at done (bastion-46c70c6bc8dbfb49bd30f57d8ec38039cf096443284e5bd4a0dac8a55ac8737d.js:28307)
    at completeRequest (bastion-46c70c6bc8dbfb49bd30f57d8ec38039cf096443284e5bd4a0dac8a55ac8737d.js:28513)
```

It's hitting it at the interpolateIfNeeded call. Since context isn't in options, it's undefined. I can jury-rig a fix using a messy if statement, but that doesn't seem to be the best approach. Let me know if that's correct.

Reply:
Error above came while creating a new repository in the products. Deleting repositories in a product seems to work fine.

Reply:
@johnsonm325 can you push up the code?